data-raw/qc-markdowns/rst/deer-creek/deer_2022_2023_qc.Rmd

```diff
@@ -0,0 +1,156 @@
+---
+title: "deer_2022_2023_qc.Rmd"
+output: html_document
+date: "2023-10-27"
+---
+
+```{r, include = F}
+library(dtplyr)
+library(data.table)
+library(tidyverse)
+library(lubridate)
+library(googleCloudStorageR)
+library(knitr)
+library(hms)
+
+root.dir <- rprojroot::find_rstudio_root_file()
+knitr::opts_knit$set(root.dir)
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r, include = F}
+gcs_auth(json_file = Sys.getenv("GCS_AUTH_FILE"))
+gcs_global_bucket(bucket = Sys.getenv("GCS_DEFAULT_BUCKET"))
+f <- function(input, output) write_csv(input, file = output)
+```
+
+```{r, include = F}
+gcs_get_object(object_name = 
+              "rst/deer-creek/data-raw/raw_2022_rst_mill_liz_ashley.xlsx",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "raw_2022_rst_deer_liz.xlsx",
+               overwrite = TRUE)
+```
+
+## Catch
+
+TODO 
+- clarify non-standard species codes
+- clarify the date recorded as 12-23-2023
+- standardize lifestage codes vs. text
+- add morts to new rows
+
+```{r, include = F}
+catch_liz <- readxl::read_excel("raw_2022_rst_deer_liz.xlsx", sheet = "Catch") |> 
+  mutate(is_plus_count = ifelse(is_plus_count == "T", T, F))
+```
+
+```{r}
+catch_clean <- catch_liz |> 
+  # standardize chinook - others can worry about later
+  mutate(species = ifelse(species == "chisal", "chinook salmon", species),
+         count = ifelse(!is.na(fork_length) & is.na(count), 1, count),
+         lifestage = str_to_lower(lifestage),
+         lifestage = case_when(lifestage %in% c("s (yearling)", "s") ~ "smolt",
+                               lifestage == "sp" ~ "silvery parr",
+                               lifestage == "sf" ~ "sac fry",
+                               lifestage == "f" ~ "fry",
+                               lifestage == "p" ~ "parr",
+                               TRUE ~ lifestage),
+         stream = ifelse(!is.na(date), "deer creek", stream),
+         is_plus_count = ifelse(is.na(is_plus_count) & 
+                                  !is.na(count), F, is_plus_count)) |> 
+  select(-c(ad_clip)) |> 
+  glimpse()
+```
+
+### simple checks: stream, date
+
+```{r}
+unique(catch_clean$stream)
+range(catch_clean$date)
+```
+
+### species
+
+```{r}
+catch_clean |> 
+  group_by(species) |> 
+  summarize(count = sum(count))
+```
+
+### fork length
+
+```{r}
+catch_clean |> 
+  filter(species == "chinook salmon") |> 
+  group_by(fork_length) |> 
+  summarize(count = sum(count)) |> 
+  ggplot(aes(x = fork_length, y = count)) +
+  geom_point()
+```
+
+```{r}
+catch_clean |> 
+  filter(species == "chinook salmon") |> 
+  group_by(date) |> 
+  summarize(fork_length = mean(fork_length, na.rm = T)) |> 
+  ggplot(aes(x = date, y = fork_length)) +
+  geom_point()
+```
+### lifestage
+
+```{r}
+catch_clean |> 
+  group_by(lifestage) |> 
+  summarize(count = sum(count))
+```
+
+##  Recapture
+
```

Comment:
Flagging this to check with Ryan - I thought they did releases and recaptures last year.